### PR TITLE
Adapt IDFF to new IDSearch, add DELTA and DELTAIDFF devices

### PR DESCRIPTION
**Pull request by xresende**

### CODE REVIEW

**Review by anacso17 — APPROVED** [review]

**Review by fernandohds564 — APPROVED** [review]

File: siriuspy/siriuspy/devices/ids.py [hunk 1]
```diff
@@ -834,6 +864,122 @@ def cmd_clear_error(self):
...
+    def pparameter_min(self):
+        """Return ID pparameter lower control limit [mm]."""
+        ctrlvars = self.pv_ctrlvars('PolShift-Mon')
+        return ctrlvars['lower_ctrl_limit']
+
+    @property
+    def pparameter_max(self):
+        """Return ID pparameter upper control limit [mm]."""
+        ctrlvars = self.pv_ctrlvars('PolShift-Mon')
+        return ctrlvars['upper_ctrl_limit']
```
Comment (murilobalves) [inline siriuspy/siriuspy/devices/ids.py:929]: why the min and max of pparameters were defined as two properties but the speed max lims as a single property returning a list with two values? It seems to break some pattern.

File: siriuspy/siriuspy/devices/ids.py [hunk 2]
```diff
@@ -834,6 +864,122 @@ def cmd_clear_error(self):
...
+    def kparameter_min(self):
+        """Return ID kparameter lower control limit [mm]."""
+        ctrlvars = self.pv_ctrlvars('Shift-SP')
+        return ctrlvars['lower_ctrl_limit']
+
+    @property
+    def kparameter_max(self):
+        """Return ID kparameter upper control limit [mm]."""
+        ctrlvars = self.pv_ctrlvars('Shift-SP')
+        return ctrlvars['upper_ctrl_limit']
```
Comment (murilobalves) [inline siriuspy/siriuspy/devices/ids.py:975]: I don't see the properties 'Shift-RB' and 'Shift-SP' in the PROPERTIES_DEFAULT of DELTA device nor the ID device from which DELTA is derived. Aren't the properties 'GainShift-RB' and 'GainShift-SP'?

**Review by murilobalves — CHANGES_REQUESTED** [review]

**Review by RafaelLyra8 — APPROVED** [review]

File: siriuspy/siriuspy/devices/ids.py [hunk 3]
```diff
@@ -950,34 +998,118 @@ def kparameter_speed_max_lims(self):
...
+    def cmd_move_start(self, timeout=None):
+        """Command to start movement."""
+        success = True
+        success &= self.cmd_move_pparam_start(timeout=timeout)
+        success &= self.cmd_move_kparam_start(timeout=timeout)
+        return success
+
+    def cmd_move_pparam_start(self, timeout=None):
+        """Command to start Pparameter movement."""
+        return self._move_start('ChangePparam-Cmd', timeout=timeout)
```
Comment (anacso17) [inline siriuspy/siriuspy/devices/ids.py:1044]: typo: ChangePparam -> ChangePParam

File: siriuspy/siriuspy/devices/ids.py [hunk 4]
```diff
@@ -874,15 +882,17 @@ class DEVICES:
...
         'MaxVelo-SP', 'MaxVelo-RB',
-        'PolModeVelo-SP', 'PolModeVelo-RB',
-        'PolModeAcc-SP', 'PolModeAcc-RB',
-        'GainModeVelo-SP', 'GainModeVelo-RB',
-        'GainModeAcc-SP', 'GainModeAcc-RB',
+        'ParkedPparameter-Cte',
+        'PParam-SP', 'PParam-RB', 'PParam-Mon',
+        'PParamVelo-SP', 'PParamVelo-RB',
+        'PParamAcc-SP', 'PParamAcc-RB',
+        'ParkedKparameter-Cte',
```
Comment (anacso17) [inline siriuspy/siriuspy/devices/ids.py:892]: typo? ParkedKparameter -> ParkedKParameter

File: siriuspy/siriuspy/devices/ids.py [hunk 5]
```diff
@@ -874,15 +882,17 @@ class DEVICES:
...
-        'Pol-Mon',
+        'Pol-Sel', 'Pol-Sts', 'Pol-Mon',
         'ChangePol-Cmd',
-        'GainShift-SP', 'GainShift-RB', 'GainShift-Mon',
         'MaxVelo-SP', 'MaxVelo-RB',
-        'PolModeVelo-SP', 'PolModeVelo-RB',
-        'PolModeAcc-SP', 'PolModeAcc-RB',
-        'GainModeVelo-SP', 'GainModeVelo-RB',
-        'GainModeAcc-SP', 'GainModeAcc-RB',
+        'ParkedPparameter-Cte',
```
Comment (anacso17) [inline siriuspy/siriuspy/devices/ids.py:888]: typo? ParkedPparameter -> ParkedPParameter

File: siriuspy/siriuspy/devices/ids.py (hunk 1 above)
Comment (xresende) [reply siriuspy/siriuspy/devices/ids.py]: deprecated

File: siriuspy/siriuspy/devices/ids.py (hunk 2 above)
Comment (xresende) [reply siriuspy/siriuspy/devices/ids.py]: deprecated

File: siriuspy/siriuspy/devices/ids.py [hunk 6]
```diff
@@ -950,34 +998,118 @@ def kparameter_speed_max_lims(self):
...
+        success &= self.cmd_move_kparam_start(timeout=timeout)
+        return success
+
+    def cmd_move_pparam_start(self, timeout=None):
+        """Command to start Pparameter movement."""
+        return self._move_start('ChangePparam-Cmd', timeout=timeout)
+
+    def cmd_move_kparam_start(self, timeout=None):
+        """Command to start Kparameter movement."""
+        return self._move_start('ChangeKparam-Cmd', timeout=timeout)
```
[the remark this answers is not in the document]
Comment (xresende) [reply siriuspy/siriuspy/devices/ids.py]: deprecated

File: siriuspy/siriuspy/devices/ids.py (hunk 3 above)
Comment (xresende) [reply siriuspy/siriuspy/devices/ids.py]: deprecated

**Review by anacso17 — APPROVED** [review]

**Pull request merged**